packages/govern-console/src/containers/DAOSettings/DAOSettings.tsx

```diff
@@ -316,30 +319,24 @@ const DaoSettings: React.FC<DaoSettingFormProps> = () => {
                 </Grid>
               </Box>
             </div>
-            <Controller
-              name="daoConfig.executionDelay"
-              control={control}
-              defaultValue={''}
-              rules={{
-                required: 'This is required.',
-                validate: (value) => positiveNumber(value),
-              }}
-              render={({ field: { onChange, value }, fieldState: { error } }) => (
-                <TextInput
-                  title={'Execution Delay'}
-                  subtitle={
-                    'Amount of time any action in your DAO will be available to be challenged before bein executed'
-                  }
-                  type="number"
-                  onChange={onChange}
-                  value={value.toString()}
-                  wide
-                  placeholder={'350s'}
-                  status={!!error ? 'error' : 'normal'}
-                  error={error ? error.message : null}
-                />
-              )}
-            />
+
+            {ipfsRulesLoading ? (
+              <ANCircularProgressWithCaption
+                caption="Fetching Execution delay from IPFS"
+                state={CircularProgressStatus.InProgress}
+              />
+            ) : (
+              <TimeInterval
+                title="Execution delay"
+                subtitle="Amount of time any transaction in your DAO will be available to be disputed by your members before being executed."
+                placeholder={'Amount'}
+                inputName="delayInputValue"
+                dropdownName="delaySelectedIndex"
+                resultName="daoConfig.executionDelay"
```

Comment:
I think we don't need to pass inputName, dropdownName, resultName to TimeInterval.. I think TimeInterval should create a unique name for those fields and not let caller determine that.. am I missing something?

Reply:
we need to pass them as those names are the name of the variable that formProvider provides
we did it the same way for ipfs input
this is the case of nested forms